### PR TITLE
Some polishing and additions to testing

### PR DESCRIPTION
**Pull request by bennybp**
# Description

Polish some testing stuff. Hopefully not too extreme or controversial. Comments welcome.

* Some weirdness with installing to the build directory is fixed
* Tests are no longer modules
* Tests are run via the RunTest script (currently in pulsar/testing, but could be moved). This script imports pulsar (easy, since it knows its location), removing the need for the TestFxns script.
* The RunTest script is installed, so it should be usable by another project (ie, StandardModules)
* The Tester is renamed to CppTester and PyTester, and the functions renamed to something more descriptive
  * When testing for the return value (via test_return), an exception will always result in a failure
* Some basic module loading/unloading/calling is tested
* Vestigial testing exports are removed 

The tests themselves remain largely the same.

## Other notes
TestModules can be removed from StandardModules now

## C++ Related Conventions
- [X] Your code is const correct
- [X] Your code is in a namespace

### CODE REVIEW

File: pulsar/testing/CppTester.hpp [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+     * \param[in] desc description of the test we are running
+     * \param[in] should_pass True if we expect the test to match expected
+     * \param[in] expected the expected return of func
+     * \param[in] func Pointer to a class member function
+     * \param[in] objptr Pointer to a class object (with which to call \p func)
+     * \param[in] args The arguments to pass to \p func
+     */
+    template<typename R1, typename R2, typename Class,
+             typename... Targs1, typename... Targs2>
+    void test_member_return(const std::string & desc,
```
Comment (ryanmrichard) [inline pulsar/testing/CppTester.hpp]: This is sexy.

**Comment by ryanmrichard** [comment]
I piggy backed on your PR, hope you don't mind

File: pulsar/testing/CppTester.hpp (hunk 1 above)
Comment (bennybp) [inline pulsar/testing/CppTester.hpp]: I can't tell if if you're being sarcastic. I think I've been programming C++ for too long

Comment (ryanmrichard) [inline pulsar/testing/CppTester.hpp]: Yes and no.  I thought it was clever, but I also realize that it's ugly by most opinions.

Comment (bennybp) [inline pulsar/testing/CppTester.hpp]: Yeah. The syntax is a mess. I expect your method of using std::bind to be more common, but this is there if you are only doing a test once or twice.

Comment (ryanmrichard) [inline pulsar/testing/CppTester.hpp]: I wonder if there's away to automate the std::bind by using auto and decltype

Comment (bennybp) [inline pulsar/testing/CppTester.hpp]: I'm not sure what you mean by automating std::bind. What do you have in mind?

Comment (ryanmrichard) [inline pulsar/testing/CppTester.hpp]: Never mind, it was me burned out on debugging.  Basically, I was theorizing about how to get the syntax you have already, things like:

~~~{.cpp}
//From TestMathSet.cpp
tester.test_return("count index false",True,False,S2.count_idx,15);
~~~
by using `auto` and `decltype` to fill in the details of `std::bind` for the member function.

**Pull request merged**